### ISSUE TITLE
使用qwen32b-vl训练grounding发现连基本的格式遵循都做不到

### ISSUE THREAD

**Issue by 1518630367** [modelscope/ms-swift#3746]
脚本如下：
CUDA_VISIBLE_DEVICES=1,0 \
MAX_PIXELS=4194400 \
nohup swift sft \
    --model /home/wumao/.cache/modelscope/hub/Qwen/Qwen2.5-VL-32B-Instruct \
    --dataset '/home/wumao/pptx_gui_agent/stage_1/all_sft_data/all_sft_data_only_pywinauto_different_pixel.jsonl' \
    --train_type lora \
    --torch_dtype bfloat16 \
    --num_train_epochs 3 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-4 \
    --lora_rank 32 \
    --lora_alpha 64 \
    --target_modules all-linear \
    --freeze_vit true \
    --gradient_accumulation_steps 16 \
    --save_strategy 'epoch' \
    --split_dataset_ratio 0 \
    --save_total_limit 3 \
    --logging_steps 5 \
    --max_length 2048 \
    --output_dir output \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    &> train.log &

**Comment by 1518630367** [comment]
self.model = Qwen2_5_VLForConditionalGeneration.from_pretrained(
            model_name_or_path,
            device_map="auto",
            trust_remote_code=True,
            torch_dtype=torch.bfloat16,
            quantization_config=bnb_config,
            attn_implementation="flash_attention_2"
        ).eval()

        if lora_path:
            print("加载lora")
            self.model = Swift.from_pretrained(self.model,lora_path)   


     messages = [
            {
                "role": "system",
                "content": [
                    {"type": "text", "text": system_prompt},
                ],
            },
            {
                "role": "user",
                "content": [
                    {
                        "type": "image",
                        "image": image_path,
                    },
                    {"type": "text", "text": full_prompt},
                ],
            }
        ]
        print(full_prompt)
        # Preparation for inference
        text_input = self.processor.apply_chat_template(
            messages, tokenize=False, add_generation_prompt=True
        )
        resize_radio = []
        image_inputs, video_inputs = process_vision_info(messages)
        for image_input in  image_inputs:
            x_radio = img_size[0] / image_input.size[0]
            y_radio = img_size[1] / image_input.size[1]
            resize_radio.append([x_radio,y_radio])
            print(image_input.size)
            print("Recover")
            print( image_input.size[0] * x_radio, image_input.size[1] * y_radio)
        # print(image_inputs)
        inputs = self.processor(
            text=[text_input],
            images=image_inputs,
            videos=video_inputs,
            padding=True,
            return_tensors="pt",
        )
        inputs = inputs.to("cuda")
        generated_ids = self.model.generate(**inputs)

**Comment by 1518630367** [comment]
> 是如何推理的

       self.model = Qwen2_5_VLForConditionalGeneration.from_pretrained(
        model_name_or_path,
        device_map="auto",
        trust_remote_code=True,
        torch_dtype=torch.bfloat16,
        quantization_config=bnb_config,
        attn_implementation="flash_attention_2"
    ).eval()

    if lora_path:
        print("加载lora")
        self.model = Swift.from_pretrained(self.model,lora_path)   


 messages = [
        {
            "role": "system",
            "content": [
                {"type": "text", "text": system_prompt},
            ],
        },
        {
            "role": "user",
            "content": [
                {
                    "type": "image",
                    "image": image_path,
                },
                {"type": "text", "text": full_prompt},
            ],
        }
    ]
    print(full_prompt)
    # Preparation for inference
    text_input = self.processor.apply_chat_template(
        messages, tokenize=False, add_generation_prompt=True
    )
    resize_radio = []
    image_inputs, video_inputs = process_vision_info(messages)
    for image_input in  image_inputs:
        x_radio = img_size[0] / image_input.size[0]
        y_radio = img_size[1] / image_input.size[1]
        resize_radio.append([x_radio,y_radio])
        print(image_input.size)
        print("Recover")
        print( image_input.size[0] * x_radio, image_input.size[1] * y_radio)
    # print(image_inputs)
    inputs = self.processor(
        text=[text_input],
        images=image_inputs,
        videos=video_inputs,
        padding=True,
        return_tensors="pt",
    )
    inputs = inputs.to("cuda")
    generated_ids = self.model.generate(**inputs)

**Comment by 1518630367** [comment]
7b是正常的

**Comment by z0o0ey** [comment]
please retry with latest swift and vllm, feel free to create a new issue if it the problem persists.
